### DISCUSSION TITLE
DietPi-Software | PowerShell

### DISCUSSION THREAD

**Discussion by Gill-Bates** [MichaIng/DietPi#5456] [Software Request]
What's about adding PowershellCore to the Dietpi-Software Catalog?

```bash
wget https://github.com/PowerShell/PowerShell/releases/download/v7.0.4/powershell-7.0.4-linux-arm32.tar.gz
mkdir ~/powershell; tar -xvf ./powershell-7.0.4-linux-arm32.tar.gz -C ~/powershell
export DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1
~/powershell/pwsh
```

**Voilá:**

![image](https://user-images.githubusercontent.com/25774289/105607439-05989100-5d9f-11eb-926a-2a78d339cae7.png)

### ADMIN EDIT: Vote for this implementation via ⬆️ button at this comment.

**Comment by MichaIng** [comment]
Many thanks for your request.

There are even amd64 Debian packages available, but to keep it simple I'd use the binary archives for all architectures.

May I ask what the use case is to use the Windows shell on Linux? I mean all instructions to do anything on a Linux system assume that the current shell is bash or bourne shell, so I cannot currently think of another reason than that the syntax or features of PowerShell are superior for certain console tasks, at least based on personal preferences 🙂. Or do users get more used to it when using PowerShell within WSL?

**Comment by Gill-Bates** [comment]
Microsoft is working to make PowerShell platform agnostic. Even though Powershell is at home in the Windows world, it can make sense to run Powershell on Linux.

Use Case:
- Running my backup scripts from Dietpi without redesigning into Bash.
- Terraform

**Comment by MichaIng** [comment]
But how or why did you run DietPi backup scripts in PowerShell in the first place? Of course it's possible when it runs on Linux but if someone checks docs or forum or anything else for backup scripts, all will be bash/bourne shell scripts.

Do you mean https://www.terraform.io/ (which works on regular Linux shells as well) or "terraforming" the usage of PowerShell on UNIX/Linux distros? 😃

But do not understand me wrong, it's great to have alternatives and help making cross-platforms work easier via cross-platform tools. I'm just interested what the initial intention in this case is where it would not come to my mind to write scripts for UNIX or GNU/Linux systems in PowerShell or batch, or scripts for Windows systems in UNIX shells.

**Comment by Josverl** [comment]
I use powershell commonly for cross-platform CI/CD scripts where the same script can run across linux/mac/windows 
bash/sh/zsh do not offer that portability , or even if the shell does, it still relies on system utilities that are not even the same across linux distros. (hence rimraf and similar workarounds) 
So that leaves either Python or pwsh , and Python requires a lot of additional processing (subprocess.run) compared to any of the shells.
( oh , and using DietPi as a self-hosted GHA runner works like a charm)

**Comment by MichaIng** [comment]
I migrated this to discussions, so you can vote with the ⬆️ button at OP post. Many thanks for your input, cross platform scripts are indeed an argument.

**Reply by echtfrank** [reply]
I'm using DietPi on my Odroid NAS / Home Server for a few years now and I am quite happy.
I am now in the process of testing / validating which Linux distro to install on my laptop to replace Windows totally.
Since my dietpi-server just n eeds to run smoothly, I am not 'bashing' a lot, and I don't need to, because it just works ;-).

Nevertheless, that changes when I use Linux on the desktop, so I was already looking to learn bash... it takes a while. I already knew that Powershell was available on Linux and asumed it was only in WSL2 on Windows, so that seemed academic for me.

Until now, because I stumbled on this interesting (old) [article](https://www.starkandwayne.com/blog/i-switched-from-bash-to-powershell-and-its-going-great/index.html) that explains a lot to me.

EDIT:
I up-voted thsi topic, because I decided that for me it is safer and quicker to use Powershell as my 'personal universal scripting environment'. Simply because I am already familiair with that.
/EDIT

Latest install set for Debian can be found [here](https://learn.microsoft.com/nl-nl/powershell/scripting/install/install-debian?view=powershell-7.3).

Sincerely, Frank

**Comment by Josverl** [comment]
@Gill-Bates ,

can you verify if your script is still valid ?
It does not appear to work for me , possibly im missing a prereq ?
